### PR TITLE
Bug 1988903: Kms details empty in only MCG deployment

### PR DESCRIPTION
**Pull request by afreen23**
- Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1988903
 - updates request execution
 - fixes the hidden token field in kms form
Screenshot from 2021-08-24 08-26-35
Screenshot from 2021-08-24 08-24-49
image

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/payloads.ts [hunk 1]
```diff
@@ -91,29 +126,26 @@ export const createStorageClusterPayload = (state: WizardState): Payload => {
...
     if (!node.labels?.[cephStorageLabel])
       requests.push(k8sPatchByName(NodeModel, node.name, null, patch));
   });
-  return requests;
+  return Promise.all(requests);
 };
+
+export const createExternalSubSystem = (subSystemPayloads: Payload[]) =>
+  Promise.all(
+    subSystemPayloads.map(async (payload) => k8sCreate(payload.model as K8sKind, payload.payload)),
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/payloads.ts:150]: Why r we using `async` when there is no `await` ?

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx [hunk 2]
```diff
@@ -86,72 +87,93 @@ const canJumpToNextStep = (name: string, state: WizardState, t: TFunction) => {
...
+const handleBackingStorageNext = async (
+  backingStorage: WizardState['backingStorage'],
+  handleError: (err: string) => void,
+  storageSystems: StorageSystemKind[] = [],
+  moveToNextStep: () => void,
+) => {
+  const { externalStorage, type } = backingStorage;
+  const { model, displayName } = getExternalStorage(externalStorage) || {};
+  const isSSPresent = storageSystems.find((ss) => ss.kind === model.kind);
   const isRhcs = externalStorage === OCSServiceModel.kind;
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx:109]:
```suggestion
  const isSSPresent = storageSystems.find((ss) => ss?.kind === model?.kind);
  const isRhcs = externalStorage === OCSServiceModel?.kind;
```

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx [hunk 3]
```diff
@@ -86,72 +87,93 @@ const canJumpToNextStep = (name: string, state: WizardState, t: TFunction) => {
...
+      await createStorageSystem(OCS_INTERNAL_CR_NAME, STORAGE_CLUSTER_SYSTEM_KIND);
+    } else if (type === BackingStorageType.EXISTING || type === BackingStorageType.LOCAL_DEVICES) {
+      await labelOCSNamespace();
+      await labelNodes(nodes);
+      if (encryption.advanced) await Promise.all(createClusterKmsResources(kms));
+      await createStorageSystem(OCS_INTERNAL_CR_NAME, STORAGE_CLUSTER_SYSTEM_KIND);
+      await createStorageCluster(state);
+    } else if (type === BackingStorageType.EXTERNAL) {
+      const { createPayload, model, displayName } = getExternalStorage(externalStorage) || {};
+      const isRhcs = externalStorage === OCSServiceModel.kind;
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx:151]:
```suggestion
      const isRhcs = externalStorage === OCSServiceModel?.kind;
```

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx:151]: not required. Its a constant object

Comment (SanjalKatiyar) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx:151]: ahh... ack !!

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx]: Actually it should be ss.spec.kind. Thanks for pointing.

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/footer.tsx]: Fixed

File: frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx [hunk 4]
```diff
@@ -79,50 +85,63 @@ export const ReviewAndCreate: React.FC<ReviewAndCreateProps> = ({ state }) => {
...
+            {t('ceph-storage-plugin~Zone: {{zoneCount, number}} zone', {
+              zoneCount: zones.size,
+              count: zones.size,
+            })}
+          </ListItem>
+        </ReviewItem>
+      )}
+      {isMCG ? (
+        <ReviewItem title={t('ceph-storage-plugin~Security')}>
+          <ListItem>{t('ceph-storage-plugin~Encryption: Enabled')}</ListItem>
```
Comment (SanjalKatiyar) [inline frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:117]: If encryption is always `Enabled` for MCG shouldn't the "Enable Encryption" checkbox in previous "Security" step should always be checked+disabled as well ?? (User can further decide whether to check KMS setting or not).
Wdyt @afreen23  ??

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:117]: yes it should be. I guess I missed that.

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/create-storage-system/create-storage-system-steps/review-and-create-step/review-and-create-step.tsx:117]: Screenshot from 2021-08-24 17-26-22

**Pull request merged**